Sources/Utils.php

```diff
@@ -2428,94 +2428,81 @@ public static function obExit(?bool $header = null, ?bool $do_footer = null, boo
 	}
 
 	/**
-	 * Parses $input to find some sort of callable.
+	 * Parses the given input to determine if it represents a callable entity.
 	 *
-	 * If a method is found, it looks for a "#" which indicates SMF should
-	 * create a new instance of the given class.
+	 * This method supports various formats of callables, including closures,
+	 * callable arrays, static methods, and class methods with optional
+	 * instance creation.
 	 *
-	 * ADD MORE HERE.
+	 * - If a class method is specified with a "#", it attempts to create
+	 *   a new instance of the class.
+	 * - If a static method is specified, it validates the method is callable.
+	 * - If input is a closure or callable array, it checks its validity.
+	 * - Plain functions are validated as callable.
+	 * - Objects themselves are not accepted as callables.
 	 *
-	 * @param mixed $input Input to parse to find a callable.
-	 * @return mixed Either a callable, or false on failure.
+	 * @param string|array $input Input to parse as a callable.
+	 * @param bool|null $ignore_errors Optional. Whether to suppress errors if the callable is invalid. Defaults to the value of `Utils::$context['ignore_hook_errors']`.
+	 *
+	 * @return callable|false Returns the callable if valid, or false on failure.
 	 */
-	public static function getCallable(mixed $input, ?bool $ignore_errors = null): mixed
+	public static function getCallable(string|array $input, ?bool $ignore_errors = null): callable|false
```

Comment:
Is callable a supported return type in PHP 8.0? (I can't check at the moment.)

Reply:
Yes; IIRC it was introduced in 7.0 along with the basic scalar types.

Reply:
I believe so and all checks pass.  Perhaps it is the first class callable syntax that you are thinking of which was 8.1: https://www.php.net/manual/en/functions.first_class_callable_syntax.php

Reply:
Yes, thanks.